### DISCUSSION TITLE
IOS first notification which is waking up app, is not triggering Background handler

### DISCUSSION THREAD

**Discussion by davx1992** [invertase/react-native-firebase#4141] [Q&A] [answered]
Hello,
I have one question regarding logic how it should work. As per instructions, when background message notification received on IOS, app is woken up, and Background handler should be triggered. Meanwhile, I noticed, that when app is killed, and new notification is received, it is waking up the app, but handler is not triggered, and when you send next notification then it works as expected, and handler is triggered. As I understood, then app after it is waken up, is in Suspended state, and in this state I handler is triggered.
On this I have a question, does it suppose to work this way or there is an issue?

**Comment by compojoom** [comment, marked as answer]
Check this out: https://github.com/invertase/react-native-firebase/issues/4104 and make the Obj-C changes. If it works for you afterwards, please write it in the issue.

**Reply by davx1992** [reply]
To which comment you are referring? I see there only Js code mentioned, not Obj-C.

**Reply by compojoom** [reply]
Read my post there again.
It talks about this file:
https://github.com/invertase/react-native-firebase/blob/master/packages/messaging/ios/RNFBMessaging/RNFBMessaging%2BAppDelegate.m#L122

You need to change the `2 *` to `8 *`

**Reply by davx1992** [reply]
Many Thanks, your approach helped!
